### ISSUE TITLE
Support devmode networks

### ISSUE THREAD

**Issue by jdtzmn** [algorand/sandbox#62]
## Problem

While it is possible to start a devmode local network by copying a `network_template.json` into the `images/algod` directory, it would be nice if there was a `config.devmode` file available so starting a local devmode network would be as simple as `./sandbox up devmode`.

## Solution

Include a `devmode_template.json` network template in the `images/algod` directory and add a `config.devmode` file similar to `config.release` but with a different `NETWORK_TEMPLATE`.

## Dependencies

None.

## Urgency

Not urgent. Would be a nice improvement.

**Comment by ipaleka** [comment]
What should be done to successfully start the indexer to? By following your solution, the following error is raised: 
```
indexer - health
{
  "data": {
    "migration-required": false
  },
  "db-available": true,
  "errors": [
    "fetcher error: HTTP 500: {\"message\":\"ledger does not have entry 1 (latest 0, committed 0)\"}\n"
  ],
  "is-migrating": false,
  "message": "0",
  "round": 0
}

```

**Comment by ipaleka** [comment]
I had to increase timeout for the Indexer to exactly 61 seconds with the solution you've provided. Any thoughts on this as for the regular release a timeout of 5 seconds is enough.

**Comment by winder** [comment]
@ipaleka what timeout are you referring to?

**Comment by ipaleka** [comment]
The time needed for a transaction created by node to appear in the Indexer database.

**Comment by algochoi** [comment]
#70 should resolve this issue -- `config.dev` now references the dev mode network template so `./sandbox up dev` should now use the dev mode network to instantly generate blocks per round rather than per ~4.5s. But I haven't checked yet if this affects the indexer timeout.

**Comment by ipaleka** [comment]
I'm afraid the timeout is still set exactly to 60 seconds.

Give it a try: [this is a repository](https://github.com/ipaleka/algorand-contracts-testing) you can use:
```
git clone -b devmode https://github.com/ipaleka/algorand-contracts-testing.git
python3 -m venv test
source test/bin/activate
cd algorand-contracts-testing/
pip install -r requirements.txt
pytest -v -k test_split_contract_transaction
```
That last line will run the test and it will pass, but if you change the INDEXER_TIMEOUT value in [helpers module](https://github.com/ipaleka/algorand-contracts-testing/blob/main/helpers.py) from 61 to even 60 (in my case) then it will fail

**Comment by winder** [comment]
@ipaleka You could add one extra transaction at the end of your test to force a new block to be created. The way "wait-for-block" works in algod is off-by-one which has led to some weird behaviors when mixed with dev-mode. For more details there are comments in this issue: https://github.com/algorand/go-algorand/issues/2716

**Comment by ipaleka** [comment]
Thank you @winder for the explanation.

This tutorial is going to be published this week on the Developer Portal and devMode is just mentioned in it. Adding that extra transaction would imo create an unnecessary confusion to the readers.

Still, one a bugfix is created I'll update the repo.

**Comment by winder** [comment]
Closed by #70
